src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj

```diff
@@ -78,4 +78,24 @@
     </ProjectReference>
   </ItemGroup>
 
+  <Target Name="GenerateBuildMetadata" BeforeTargets="BeforeBuild"
+          Inputs="$(MSBuildProjectFile)"
+          Outputs="$(IntermediateOutputPath)Generated_BuildMetadata.cs">
+
+      <WriteLinesToFile
+        File="$(IntermediateOutputPath)Generated_BuildMetadata.cs"
+        Lines="// &lt;autogenerated /&gt;
+
+internal static class BuildMetadata
+{
+    public const string ILLinkImplAssemblyVersion = &quot;$(NetCoreAppToolCurrentVersion)&quot;%3B
```

Comment:
I think it would be better to just pass the TFM string in: `$(NetCoreAppToolCurrent`. That avoids calculating the TFM string in LinkTask.cs (in case we ever change the format).